### PR TITLE
feat: implementing pip-19

### PR DESCRIPTION
**Pull request by b00f**
## Description

This PR implements the [PIP-19](https://pips.pactus.org/PIPs/pip-19)

### CODE REVIEW

File: state/score/score.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		}
+
+		delete(sm.certs, oldHeight)
+	}
+}
+
+func (sm *Manager) AvailabilityScore(valNum int32) float64 {
+	data, ok := sm.vals[valNum]
+	if ok {
+		if data.inCommittee == 0 {
```
Comment (Ja7ad) [inline state/score/score.go:65]: Impossible inCommittee = 0 and absent = n?

File: network/relay_service.go [hunk 2]
```diff
@@ -76,7 +76,8 @@ func (rs *relayService) checkConnectivity() {
 				rs.logger.Info("try connecting relay node", "addr", ai.Addrs)
 				err := ConnectSync(rs.ctx, rs.host, ai)
 				if err != nil {
-					rs.logger.Warn("unable to connect to relay node", "error", err, "addr", ai.Addrs)
+					// TODO: Make me Warn?
+					rs.logger.Debug("unable to connect to relay node", "error", err, "addr", ai.Addrs)
```
[the remark this answers is not in the document]
Comment (kehiy) [reply network/relay_service.go:80]: I think it's better to keep it as debug. it's just a failing connection to one of many relays, so having a warn log each time can make redundant data when it's not in development and debuging phase.

[the remark this answers is not in the document]
Comment (b00f) [reply network/relay_service.go:80]: Yes, and users always complain about it. Until we fix the relay issue, right now, we don't have any traffic on our relay nodes, even though nodes are connected to them.

@ali-mahdavi99 @kehiy @Ja7ad any idea?

File: state/score/score.go (hunk 1 above)
Comment (b00f) [reply state/score/score.go:65]: No, You should be in committee then not vote (or absent).

File: network/relay_service.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ja7ad) [reply network/relay_service.go:80]: I think the debug level is good, we need traffic for decide.

**Pull request merged**